scenarios/aws/ec2/os_win.go

```diff
@@ -3,25 +3,59 @@ package ec2
 import (
 	"fmt"
 	"os"
+	"strings"
+
+	componentsos "github.com/DataDog/test-infra-definitions/components/os"
 )
 
 func getWindowsOpenSSHUserData(publicKeyPath string) (string, error) {
 	publicKey, err := os.ReadFile(publicKeyPath)
 	if err != nil {
 		return "", err
 	}
+	return buildAWSPowerShellUserData(
+			componentsos.SetupSSHScriptContent,
+			windowsPowerShellArgument{name: "authorizedKey", value: string(publicKey)},
+		),
+		nil
+}
+
+type windowsPowerShellArgument struct {
+	name  string
+	value string
+}
+
+func (a windowsPowerShellArgument) String() string {
+	return fmt.Sprintf("-%s %s", a.name, a.value)
+}
+
+func buildAWSPowerShellUserData(scriptContent string, arguments ...windowsPowerShellArgument) string {
+	scriptLines := strings.Split(scriptContent, "\n")
+	userDataLines := make([]string, 0, len(scriptLines)+6+len(arguments))
+	userDataLines = append(userDataLines, "<powershell>")
+	for _, line := range scriptLines {
+		// indent script lines by one tab
+		userDataLines = append(userDataLines, fmt.Sprintf("		%s", line))
+	}
+	userDataLines = append(userDataLines, "</powershell>")
+	userDataLines = append(userDataLines, "<persist>true</persist>")
```

Comment:
Note for reviewers:

Using `persist: true` option on aws to allow calling user data on each instance reboot, I noticed without this the userdata on custom amis did not seem to be called.